sql/src/main/java/io/crate/planner/consumer/ReduceOnCollectorGroupByConsumer.java

```diff
@@ -150,13 +154,31 @@ private AnalyzedRelation optimizedReduceOnCollectorGroupBy(QueriedTable table, T
                 table.tableRelation().validateOrderBy(orderBy);
             }
 
+            PlanNodeBuilder.GroupByCollectNodeInfo groupByCollectNodeInfo;
+            if(context.rootRelation() == table && !tableInfo.schemaInfo().systemSchema()) {
```

Comment:
maybe this whole stuff that is happening in all 3 Consumers could be put into one static method somewhere?